### PR TITLE
c++11 building support

### PR DESCRIPTION
**Pull request by wbpcode**
Signed-off-by: wangbaiping <wbphub@live.com>

```
std::make_unique (from c++14) -> absl::make_unique
std::string_view (from c++17) -> absl::string_view
std::optional (from c++17) -> absl::optional
std::nullopt (from c++17) -> absl::nullopt
```

`absl::string_view/absl::optional/absl::nullopt/...` will be alias of `std::string_view/std::optional/std::nullopt/...` after c++17 by default. So it's ok to always use the `absl::*` in c++11/c++14/c++17 versions.

### CODE REVIEW

**Comment by wu-sheng** [comment]
So, we will have c++11 and c++17 in support at the same time?

**Comment by wbpcode** [comment]
> So, we will have c++11 and c++17 in support at the same time?

yep.

**Comment by wu-sheng** [comment]
Could you add a CI for C++11 build and update the doc to mention 11 and 17 supported?

**Comment by wbpcode** [comment]
In fact it's build by c++11 by default for now because I remove the explicit option of `--cxxopt=-std:c++17`. But it's OK for me to add a new test for it.

**Comment by wbpcode** [comment]
some warnings need to be resolved.

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ![cpp2sky test](https://github.com/SkyAPM/cpp2sky/workflows/cpp2sky%20test/badge.svg)
 
-Distributed tracing and monitor SDK in CPP for Apache SkyWalking APM
+Distributed tracing and monitor SDK in CPP for Apache SkyWalking APM. This SDK is compatible with C++ 17, C++ 14, and C++ 11.
```
Comment (wu-sheng) [inline README.md:5]: 👍

**Review by wu-sheng — APPROVED** [review]
LGTM.

**Pull request merged**